dbms/src/Functions/FunctionsRegexp.h

```diff
@@ -0,0 +1,1009 @@
+                    else
+                        *has_data_type_nothing = true;
+                }
+            }
+            else
+            {
+                if (!arg->isInteger())
+                    throw Exception(fmt::format("Illegal type {} of argument of regexp function", arg->getName()), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+        }
+    }
+};
+
+// regexp and regexp_like functions are executed in this macro
+#define EXECUTE_REGEXP_LIKE()                                                                                                                       \
+    do                                                                                                                                              \
+    {                                                                                                                                               \
+        REGEXP_CLASS_MEM_FUNC_IMPL_NAME(RES_ARG_VAR_NAME, *(EXPR_PARAM_PTR_VAR_NAME), *(PAT_PARAM_PTR_VAR_NAME), *(MATCH_TYPE_PARAM_PTR_VAR_NAME)); \
+    } while (0);
+
+// Method to get actual match type param
+#define GET_MATCH_TYPE_ACTUAL_PARAM()                                                                             \
+    do                                                                                                            \
+    {                                                                                                             \
+        GET_ACTUAL_STRING_PARAM(MATCH_TYPE_PV_VAR_NAME, MATCH_TYPE_PARAM_PTR_VAR_NAME, ({EXECUTE_REGEXP_LIKE()})) \
+    } while (0);
+
+// Method to get actual pattern param
+#define GET_PAT_ACTUAL_PARAM()                                                                              \
+    do                                                                                                      \
+    {                                                                                                       \
+        GET_ACTUAL_STRING_PARAM(PAT_PV_VAR_NAME, PAT_PARAM_PTR_VAR_NAME, ({GET_MATCH_TYPE_ACTUAL_PARAM()})) \
+    } while (0);
+
+// Method to get actual expression param
+#define GET_EXPR_ACTUAL_PARAM()                                                                        \
+    do                                                                                                 \
+    {                                                                                                  \
+        GET_ACTUAL_STRING_PARAM(EXPR_PV_VAR_NAME, EXPR_PARAM_PTR_VAR_NAME, ({GET_PAT_ACTUAL_PARAM()})) \
+    } while (0);
+
+// The entry to get actual params and execute regexp functions
+#define GET_ACTUAL_PARAMS_AND_EXECUTE() \
+    do                                  \
+    {                                   \
+        GET_EXPR_ACTUAL_PARAM()         \
+    } while (0);
```

Comment:
Maybe we can combine these micros into one？

Reply:
> Maybe we can combine these micros into one？

I believe that readability is very important and it's more clear to seperate them.